sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnectionOptions.cs

```diff
@@ -148,5 +158,19 @@ public TimeSpan ConnectionIdleTimeout
         ///
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString() => base.ToString();
+
+        /// <summary>
+        ///   Creates a new copy of the current <see cref="EventHubConnectionOptions" />, cloning its attributes into a new instance.
+        /// </summary>
+        ///
+        /// <returns>A new copy of <see cref="EventHubConnectionOptions" />.</returns>
+        ///
+        internal EventHubConnectionOptions Clone()
+        {
+            var clone = EventHubConnectionOptionsExtensions.Clone(this);
```

Comment:
Can this property be baked into the Clone method?

Reply:
Not without making it public.   I'm still tooling around with the structure of things and I think I can pull this into the client directly, so this is likely to go away.

Reply:
I was able to limit the scope of where this was used enough that it doesn't have to flow with the options.   This is now just passed into the `AmqpClient` constructor.